Katello/katello
# Fixes #32277 - make org creation idempotent and async from seeding

**TODO**
- [x] new db field on org to track owner and library env creation in candlepin
- [x] actually test this
- [x] remove dead code unrelated to this pr to boost # of lines deleted
- [x] add debug cert creation to OrganizationCreator
- [x] fix tests / vcrs

**This PR**
- Adds a middleware which ensures all organizations are created as far as katello is concerned
- Removes the `Organization::Create` dynflow action (and a few related ones)
- Uses the new `OrganizationCreator` in the middleware and where the dynflow task was used
- Adds some new errors that are generally useful (for Pinging services and checking candlepin status more accurately)
- Consolidates some of the org related testing that seems to be spread around a lot
- Fires a UI notification if `create_all_organizations` fails so admins can fix the problem (candlepin down)

**to test**

- migrate the db

- tail the candlepin.log to understand the traffic pattern this generates
- hit any URL of the app multiple times
- notice the query to check non-created orgs won't be run on each request for that given rails process since the the middleware is initialized only once
-  multiple requests shouldn't cause a problem cause org create doesn't lock the org anymore
- remember that the org create API calls `OrganizationCreator.create!` whereas seeding calls `OrganizationCreator.seed!` to separate the two. this implies that this middleware should only be relevant to orgs that were seeded but not created through app usage (ie Default Organization)

To reset your orgs: `Organization.update_all(created_in_katello: false)`

- turn candlepin off `systemctl stop tomcat` or 'break' candlepin by moving /etc/candlepin/broker.xml somewhere else and doing `systemctl restart tomcat` and test some more to see the failure scenario

**try to break it so I can fix it**
